### ISSUE TITLE
Deleting messages sometimes or always does not update pager, and gets stuck

### ISSUE THREAD

**Issue by auouymous** [neomutt/neomutt#3585]
```
set mbox_type = Maildir
set folder = "~/.mail"
set spool_file = "+inbox"
set pager_index_lines = 9
set delete = yes
unset resolve
macro index,pager <delete> "<delete-message><sync-mailbox>"
```

That has always deleted the message with neomutt <=20211029 and moved pager to the next message. With 20220415 through 34816020295ace8f4b94a4da4c89367354457f47, it deletes the message, and updates the index and index status bar to the next message. However, the pager still contains the deleted message, and the pager status bar shows a 'D' for %S, while %C/%m indicates the message was deleted.

Pressing delete again does nothing, which is good because I can't accidentally delete the next message. And pressing delete a third time shows "Mailbox is unchanged", although this message is sometimes shown on the second press. Moving up or down in index, or closing and opening pager, is required to get out of this stuck state, and continue reading or deleting messages.

In my primary client this happens every 10 or so messages, but for some reason it happens almost every time on my test client with the same config.

```
macro index,pager <delete> "<delete-message><next-entry><sync-mailbox>"
```

Inserting `next-entry` seems to fix the issue, but it should probably work without this. I also have next-entry beep when used while on last message, so deleting last message now beeps with this hack.

## Steps to Reproduce

Add above config to .muttrc, fill maildir inbox with messages, run neomutt, open pager, and start deleting messages.

## NeoMutt Version

```
NeoMutt 20220429-256-348160-dirty

System: Linux 5.4.228 (i686)
ncurses: ncurses 6.3.20221203 (compiled with 6.3.20221203)

Configure options: --prefix=/usr --build=i686-pc-linux-gnu --host=i686-pc-linux-gnu --mandir=/usr/share/man --infodir=/usr/share/info --datadir=/usr/share --sysconfdir=/etc --localstatedir=/var/lib --docdir=/usr/share/doc/neomutt-99999999 --with-sysroot=/ --libdir=/usr/lib CCACHE=none CC_FOR_BUILD=i686-pc-linux-gnu-gcc --disable-doc --disable-nls --disable-notmuch --disable-autocrypt --disable-gpgme --disable-pgp --disable-smime --disable-bdb --disable-gdbm --disable-kyotocabinet --disable-qdbm --disable-tokyocabinet --disable-lz4 --enable-zlib --disable-zstd --disable-idn --disable-gss --disable-lmdb --disable-sasl --with-ui=ncurses --sysconfdir=/etc/neomutt --disable-ssl --disable-gnutls --disable-testing

Compilation CFLAGS: -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I//usr/include -DNCURSES_WIDECHAR -I//usr/include

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls -gpgme 
  -gsasl -gss -hcache -homespool -idn +inotify -locales_hack -lua -mixmaster 
  -nls -notmuch -openssl -pgp +regex -sasl -smime -sqlite +sun_attachment 

MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc/neomutt"
```

## Extra Info

* Gentoo Linux
* Were you using multiple copies of NeoMutt at once? No
* Were you using 'screen' or 'tmux'? No
* Is your email local (maildir) or remote (IMAP)? maildir

**Comment by auouymous** [comment]
I was wrong about it starting around 20220415, it was introduced in 80b28623d312693c845312bb7053b49d6e0253f0.

**Comment by auouymous** [comment]
The "fix" for this also causes another issue related to #3609.

With `pager_read_delay = 0`, `<delete-message><sync-mailbox>` deletes a message and advances to the next message. However, `<delete-message><next-entry><sync-mailbox>` deletes a message, advances to next and immediately marks it as read. I noticed this behavior today and upgraded to get #3640, but it didn't fix deleting.

Syncing after moving up or down in pager does not mark the message as read. So something in `<sync-mailbox>` causes it to happen only when deleting messages, because the next message isn't marked read until the end when manually running each command.

A non-zero `pager_read_delay` does not have this issue, before or after #3640.

**Comment by roccoblues** [comment]
Hi @auouymous, can you still reproduce this with the latest version?

**Comment by auouymous** [comment]
It appears to have been fixed. However, I am not able to bisect back to the bad commit to find the one that fixed it.
